rotkehlchen/rotkehlchen.py

```diff
@@ -173,35 +195,108 @@ def __init__(self, args: argparse.Namespace) -> None:
                         greenlet.kwargs['only_cache'] is False and
                         account_tuple in greenlet.kwargs['filter_query'].accounts
                 ):
+                    kill_greenlet_and_acquire_locks(
+                        stack=stack,
+                        acquired_by_this_method=acquired_by_this_method,
+                        greenlet=greenlet,
+                        locks=[evm_manager.transactions.address_tx_locks[address]],
+                    )
+                elif (
+                    is_evm_tx_greenlet and
+                    greenlet.args[0].__func__.__qualname__ == 'RestAPI._decode_pending_evm_transactions'  # noqa: E501
+                ):
+                    kill_greenlet_and_acquire_locks(
+                        stack=stack,
+                        acquired_by_this_method=acquired_by_this_method,
+                        greenlet=greenlet,
+                        locks=[
+                            evm_manager.transactions_decoder.undecoded_tx_query_lock,
+                            evm_manager.transactions.missing_receipts_lock,
+                        ],
+                    )
 
-                    greenlet.kill()
+            # Check related periodic tasks greenlets
+            for greenlet in self.task_manager.running_greenlets.get(self.task_manager._maybe_query_evm_transactions, []):  # noqa: E501
+                address = greenlet.kwargs.get('address')
+                if address is None or address in addresses:
+                    kill_greenlet_and_acquire_locks(
+                        stack=stack,
+                        acquired_by_this_method=acquired_by_this_method,
+                        greenlet=greenlet,
+                        locks=[evm_manager.transactions.address_tx_locks[address]],
+                    )
+
+            for greenlet in self.task_manager.running_greenlets.get(self.task_manager._maybe_decode_evm_transactions, []):  # noqa: E501
+                addresses_argument = greenlet.kwargs.get('addresses')
+                if addresses_argument is None or len(set(addresses_argument) & set(addresses)) > 0:
+                    kill_greenlet_and_acquire_locks(
+                        stack=stack,
+                        acquired_by_this_method=acquired_by_this_method,
+                        greenlet=greenlet,
+                        locks=[evm_manager.transactions.missing_receipts_lock],
+                    )
 
-            tx_query_task_greenlets = self.task_manager.running_greenlets.get(self.task_manager._maybe_query_evm_transactions, [])  # noqa: E501
-            for greenlet in tx_query_task_greenlets:
-                if greenlet.kwargs['address'] in addresses:
-                    greenlet.kill()
+            for greenlet in self.task_manager.running_greenlets.get(self.task_manager._maybe_schedule_evm_txreceipts, []):  # noqa: E501
```

Comment:
Same comment as above. You are trying to kill the scheduling event here and not the actual greenlet that will be scheduled